core/src/main/java/io/kestra/core/storages/StorageInterface.java

```diff
@@ -50,9 +54,24 @@ default boolean exists(URI uri) {
     @Retryable(includes = {IOException.class}, excludes = {FileNotFoundException.class})
     Long lastModifiedTime(URI uri) throws IOException;
 
+    @Retryable(includes = {IOException.class}, excludes = {FileNotFoundException.class})
+    Long createdAtTime(URI uri) throws IOException;
+
+    // Not sur what representation you except for permissions ?
+    @Retryable(includes = {IOException.class}, excludes = {FileNotFoundException.class})
+    Long permissions(URI uri) throws IOException;
```

Comment:
We already have methods for lastModifiedTime and file size.
Instead of adding more methods which would mean more call to the API, I wonder if we shouldn't create a metadata(UI) methods which will return a metadata object with all the information. We could deprecate the exisitngh methods and refactor them later

Reply:
Sounds like a good idea I'll change that